models/ProfileStats.js

```diff
@@ -0,0 +1,26 @@
+import mongoose from "mongoose";
+const viewSchema = new mongoose.Schema({
+  views: {
+    type: Number,
+    required: true,
+  },
+  date: {
+    type: Date,
+    required: true,
+  },
+});
+
+const profileStatsSchema = new mongoose.Schema({
+  username: {
+    type: String,
+    required: true,
+  },
+  views: {
+    type: [viewSchema],
```

Comment:
I think simplifying the model will make our code cleaner, similar to the way links are done https://github.com/EddieHubCommunity/LinkFree/blob/main/models/Link.js

Reply:
do you suggesting schema like { date, profile views} ?

Reply:
Yes I think so. Here it is formatted for others to be able to read...

```js
new mongoose.Schema({
   username: String,
   views: {
      type: Number,
      default: 0,
   },
   date:{
      type: Date,
      default: new Date(new Date().toLocaleDateString()),
   }
});
```

It is probably worth creating a relationship to the profile with

```js
   profile: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "profile",
    },
```

Reply:
Thanks for the suggestion, @eddiejaoude. Working on it